client/js/libs/handlebars/parse.js

```diff
@@ -81,5 +81,8 @@ module.exports = function parse(text) {
 		}
 
 		return fragments;
+	}).join("") + linkParts.map((part) => {
+		const escapedLink = Handlebars.Utils.escapeExpression(part.link);
+		return `<div class="preview" data-url="${escapedLink}"></div>`;
```

Comment:
I'm not 100% sure about putting these right next to the link, but I can't think of a better option. It's probably the best we can do when there are multiple previews.